pymobiledevice3/services/crash_reports.py

```diff
@@ -127,28 +128,35 @@ def watch(self, name: str = None, raw: bool = False) -> Generator[str, None, Non
                 else:
                     yield crash_report
 
-    def get_new_sysdiagnose(self, out: str, erase: bool = True) -> None:
+    def get_new_sysdiagnose(self, out: str, erase: bool = True, *, timeout: Optional[float] = None) -> None:
         """
         Monitor the creation of a newly created sysdiagnose archive and pull it
         :param out: filename
         :param erase: remove after pulling
+        :keyword timeout: Maximum time in seconds to wait for the completion of sysdiagnose archive
+            If None (default), waits indefinitely
         """
-        sysdiagnose_filename = self._get_new_sysdiagnose_filename()
+        end_time = None
+        if timeout is not None:
+            end_time = time.monotonic() + timeout
+        sysdiagnose_filename = self._get_new_sysdiagnose_filename(end_time)
         self.logger.info('sysdiagnose tarball creation has been started')
-        self._wait_for_sysdiagnose_to_finish()
+        self._wait_for_sysdiagnose_to_finish(end_time)
         self.pull(out, entry=sysdiagnose_filename, erase=erase)
 
     @staticmethod
     def _sysdiagnose_complete_syslog_match(message: str) -> bool:
         return message == 'sysdiagnose (full) complete' or 'Sysdiagnose completed' in message
 
-    def _wait_for_sysdiagnose_to_finish(self) -> None:
+    def _wait_for_sysdiagnose_to_finish(self, end_time: Optional[float]) -> None:
```

Comment:
add the optionals a `= None` so these optionals are really optional

Reply:
done, apologies for missing that.